src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java

```diff
@@ -253,8 +261,17 @@ public void validate(FailureCollector failureCollector) {
         }
       }
 
-      if (!containsMacro(SQL) && Strings.isNullOrEmpty(sql)) {
-        failureCollector.addFailure("SQL not specified. Please specify a SQL to execute", null).withConfigProperty(SQL);
+      if (!containsMacro(SQL)) {
+        if (Strings.isNullOrEmpty(sql)) {
+          failureCollector.addFailure("SQL not specified. Please specify a SQL to execute", null)
```

Comment:
failureCollector.addFailure("SQL not specified.",  "Please specify a SQL to execute.").withConfigProperty(SQL);

Reply:
PR updated